Fixes #19002 - introduced ApplicationRecord

lzap opened the pull request:
Because Rails 5 will require apps to use ApplicationRecord and because this makes things easier for us to redirect orchestration log messages from sql to orch logger, we are making a change in core and introducing this class (#13772).

Most `ActiveRecord::Base` use must be replaced, specifically:

* `app/model/` - model classes
* `db/migrate/` - "fake" reopened classes otherwise TypeError: superclass mismatch for class XYZ
* `test/` - reopened classes otherwise TypeError: superclass mismatch for class XYZ

This requires https://github.com/theforeman/foreman/pull/3729 to be merged in core first. Tests will fail.

lzap commented:
The patch in core passed review and is RTM.

lzap commented:
Guys @bbuckingham @ehelms @jlsherrill or @tbrisker this is an intrusive change in core, this patch makes sure Katello is not broken. Can you guys give me review and Ready to merge flag, then we will merge both at the same time so CI is not broken much. Thanks!

ehelms commented:
As you said, this is an intrusive change with 3 days to branch that makes me hesitent to pull in large changes. Is this targeted for 1.15?

lzap commented:
Good point, this can wait after branching sure.

lzap commented:
Added newly created `db/migrate/20170321012632_fill_in_content_view_components.rb`, rebased, since we branched off, please merge once core patch is merged. I suggest manual merge, rebase and grep for `ActiveRecord::Base` in the codebase for *relevant* occurances (if new model was added for example). There are few irrelevant items.

iNecas commented:
As we agreed on previously, the Foreman part was posponed until branching 1.15, and now it got merged: @Katello/katello-core please have a look at this PR to make Katello working with the foreman develop again.

iNecas commented:
[test]

jlsherrill commented:
ACK thanks @lzap
